### ISSUE TITLE
[Input] Email validation Input/Button combo

### ISSUE THREAD

**Issue by stevem63** [momentum-design/momentum-ui#98]
In a couple different places within Webex teams, we do input validation to check whether a string matches a regex pattern for a valid email format. Only once the text value in the <Input> component matches the regex do we enable the <Button>. It would be nice to have a component that wrapped the Input and Button components to do this sort of thing for us. Here is the regex we use:

`/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/`

## Issue Type
Feature Request

## Component or Pattern Affected

None? It's a new component

## Expected Behavior

![](https://user-images.githubusercontent.com/41303817/53613987-fa299480-3b8c-11e9-84cc-109665031228.png)
![](https://user-images.githubusercontent.com/41303817/53613990-fc8bee80-3b8c-11e9-99b6-a71fb62541f4.png)

## Possible Solution
Make a component that wraps the Button and Input components like I mentioned above and include the functionality I mentioned with a regex such as the one I listed above.
